### PR TITLE
feedback knoweldege base with thumbsUp/thumbsDown

### PR DESCRIPTION
**Pull request by Iru89**
## Description

Add feedback_knowledgebase event
Be able to add ThumbsUp and ThumbsDown buttons to give feedback on webchat messages.
When the flow builder creates a message using the knowledge base Uses the Text component with new withfeedback and inferenceid attributes

The withfeedback and inferenceid attributes cannot be declared using camelCase because when they pass through the backen they come back with everything in lowercase, it doesn't work with snake_case either. I'm not sure but I think it's because of the html parser, the <message> attributes have to be single words or use kebab-case.

## Context

Review this PR commit by commit:
[plugin-hubtype-analytics](dc8f37331340db098b60627766afe57a8f971fd6): add feedback_knowledgebase event

[botonic-react](3fa97b0398873da8a084239a7a54696fdf0142e2): Refactor the FooterMessage and add the FeedbackMessage which contains the ThumbsUp ThumbsDown buttons and uses the trackEvent function when clicked.  

[botonic-react](bb98f33a34fb2f181ac372a15d8bb8ca1ac9486f): Add the possibility to inject the onTrackEvent function from the bot to the webchat to create events from the frontend. Just like we already have the onInit, onOpen, onClose, onMessage functions.

[plugin-flow-builder](8d90eb7f416ff7f20f57f6419445c693682b0295): When the flow builder plugin uses the knowledge base to generate a message it creates a @botonic/react Text with the new withfeedback and inferenceid attributes needed for tracking.

The other commits are a refactor that renumbers the with_feedback attrbute to feebackEnabled using camelCase

## Approach taken / Explain the design

Design
Captura de pantalla 2024-06-21 a las 14 02 23

## To document / Usage example

How to pass the tracking function from bot to webchat
/webchat/index.ts
```typescript
export const webchat: WebchatArgs = {
  onInit: app => {
    window.botonicOnInit(app)
  },
  ...
  onTrackEvent: async (request, eventName, args) => {
    const pluginHubtypeAnalytics = new BotonicPluginHubtypeAnalytics()
    const htEventProps = {
      action: eventName as EventAction,
      ...args,
    } as HtEventProps
    await pluginHubtypeAnalytics.trackEvent(request, htEventProps)
  },
  theme: {
  ...
```

## Testing

Adds a test for the new feedback_knowledgebase event

### CODE REVIEW

File: packages/botonic-plugin-flow-builder/src/content-fields/flow-knowledge-base.tsx [hunk 1]
```diff
@@ -5,9 +5,10 @@ import { ContentFieldsBase } from './content-fields-base'
 import { HtKnowledgeBaseNode } from './hubtype-fields'
 
 export class FlowKnowledgeBase extends ContentFieldsBase {
-  public code = ''
+  public code: string = ''
```
Comment (vanbasten17) [inline packages/botonic-plugin-flow-builder/src/content-fields/flow-knowledge-base.tsx:8]: There is no need to type as string here as it is initialized as string

File: packages/botonic-plugin-hubtype-analytics/src/event-models/ht-event-feedback-knowledgebase.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  option: string
+  value: number
+  comment?: string
+
+  constructor(event: EventFeedbackKnowledgebase, requestData: RequestData) {
+    super(event, requestData)
+    this.type = EventType.WebEvent
+    this.action = event.action
+    this.knowledgebase_inference_id = event.knowledgebaseInferenceId
+    this.feedback_target_id = event.feedbackTargetId // ?? case_id, message_id, conversation_id ???, webview_name
```
Comment (vanbasten17) [inline packages/botonic-plugin-hubtype-analytics/src/event-models/ht-event-feedback-knowledgebase.ts:25]: Are these comments still not resolved?

File: packages/botonic-plugin-hubtype-analytics/src/event-models/ht-event-feedback-knowledgebase.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+} from '../types'
+import { HtEvent } from './ht-event'
+
+export class HtEventFeedbackKnowledgebase extends HtEvent {
+  action: EventAction.FeedbackKnowledgebase
+  knowledgebase_inference_id: string
+  feedback_target_id?: string
+  feedback_group_id?: string
+  possible_options: string[]
+  possible_values: number[]
```
Comment (vanbasten17) [inline packages/botonic-plugin-hubtype-analytics/src/event-models/ht-event-feedback-knowledgebase.ts:15]: We have to add this event in backend, but should this be optional?

File: packages/botonic-react/src/components/message/index.jsx [hunk 4]
```diff
@@ -67,8 +70,10 @@ export const Message = props => {
       typeof e === 'string' ? renderLinks(e) : e
     )
 
-  const { timestampsEnabled, getFormattedTimestamp, timestampStyle } =
-    resolveMessageTimestamps(getThemeProperty, enabletimestamps)
+  const { timestampsEnabled, getFormattedTimestamp } = resolveMessageTimestamps(
```
Comment (vanbasten17) [inline packages/botonic-react/src/components/message/index.jsx:73]: Have we removed timestampstyle?

**Review by vanbasten17 — APPROVED** [review]

File: packages/botonic-plugin-hubtype-analytics/src/event-models/ht-event-feedback-knowledgebase.ts (hunk 2 above)
Comment (Iru89) [reply packages/botonic-plugin-hubtype-analytics/src/event-models/ht-event-feedback-knowledgebase.ts]: done

File: packages/botonic-react/src/components/message/index.jsx (hunk 4 above)
Comment (Iru89) [reply packages/botonic-react/src/components/message/index.jsx:73]: is used inside the timestamp component

**Pull request merged**